scripts/migrations.js

```diff
@@ -0,0 +1,79 @@
+import Constants from '../constants.js';
+import Settings from '../settings.js';
+
+export default class Migrations {
+	constructor() {
+		this._settings = new Settings();
+	}
+	/* Migrates the description of Custom Convenient Effects to use the ActiveEffect5e#description field instead of the legacy flags.
+	 *
+	 * @param {string} id - the ID of the custom effects item
+	 * @returns {Promise} a promise that resolves when all the Item's ActiveEffects descriptions have migrated
+	 */
+	async _migrateCustomEffectsItemEffectDescriptions(id, migration) {
+		const item = this._settings._findCustomEffectsItem(id);
+		const updates = [];
+		const effects = item.effects.filter((effect) =>
+			effect.getFlag(Constants.MODULE_ID, Constants.FLAGS.HAS_MIGRATED)
+		);
+		for (const effect of effects) {
+			const effectDescription = effect.description;
+			const flagDescription = effect.getFlag(
+				Constants.MODULE_ID,
+				Constants.FLAGS.DESCRIPTION
+			);
+			const legacyDescription =
+				effect.flags.convenientDescription ??
+				effect.flags.customEffectDescription;
+			const description =
+				!!effectDescription &&
+				![
+					'Applies Custom Effect',
+					'Applies custom effects',
+					'test',
+				].some((d) => d === effectDescription)
+					? effectDescription
+					: !!flagDescription &&
+					  ![
+							'Applies Custom Effect',
+							'Applies custom effects',
+							'test',
+					  ].some((d) => d === flagDescription)
+					? flagDescription
+					: !!legacyDescription &&
+					  ![
+							'Applies Custom Effect',
+							'Applies custom effects',
+							'test',
+					  ].some((d) => d === legacyDescription)
+					? legacyDescription
+					: '';
+			updates.push({
+				_id: effect.id,
+				description,
+				[`flags.${Constants.MODULE_ID}.${Constants.FLAGS.IS_CONVENIENT}`]: true,
+				[`flags.${Constants.MODULE_ID}.${Constants.FLAGS.HAS_MIGRATED}`]:
```

Comment:
Curious if we should prefer "last migration ran" saved in the settings or something, rather than setting a new flag on every old effect.

Doing so would make this function check:

```
if ((this._settings.lastMigrationRan ?? '0.0.0') >= currentVersion) return; // don't do migration
```

It would also remove the `.filter` we do above.

That `>=` is easier said than done I suppose. In reality, it would be some sort of function to be able to compare strings like '1.2.0' and '2.0.0' and know that '2.0.0' is greater. Not too difficult, but a required thing.

Then at the end of the `execute` function, we'd just set `lastMigrationRan` on the settings to the latest version.

Reply:
If this seems like a lot, I could probably do this myself.